controller/src/beerocks/master/tasks/dynamic_channel_selection_task.cpp

```diff
@@ -26,7 +26,69 @@ dynamic_channel_selection_task::dynamic_channel_selection_task(db &database_,
 {
     m_fsm_state = eState::INIT;
 }
+beerocks::eChannelScanErrCode dynamic_channel_selection_task::dcs_request_scan_dump()
+{
+    // When a dump is requested, do not set the scan_in_progress flag
+    // simply send the request as is, and wait for the results ready event
+
+    auto request = beerocks::message_com::create_vs_message<
+        beerocks_message::cACTION_CONTROL_CHANNEL_DUMP_SCAN_RESULTS_REQUEST>(cmdu_tx);
+    if (request == nullptr) {
+        LOG(ERROR) << "Failed building message cACTION_CONTROL_CHANNEL_DUMP_SCAN_RESULTS_REQUEST";
+        return beerocks::eChannelScanErrCode::CHANNEL_SCAN_INTERNAL_FAILURE;
+    }
+
+    // get the parent node to send the CMDU to the agent
+    auto radio_mac_str = beerocks::net::network_utils::mac_to_string(m_radio_mac);
+    auto ire           = database.get_node_parent_ire(radio_mac_str);
+    son_actions::send_cmdu_to_agent(ire, cmdu_tx, database, radio_mac_str);
+
+    return beerocks::eChannelScanErrCode::CHANNEL_SCAN_SUCCESS;
+}
+beerocks::eChannelScanErrCode dynamic_channel_selection_task::dcs_request_scan_trigger()
+{
+    // When a scan is requested send the scan parameters Channel pool & Dwell time
 
+    auto request = beerocks::message_com::create_vs_message<
+        beerocks_message::cACTION_CONTROL_CHANNEL_SCAN_TRIGGER_SCAN_REQUEST>(cmdu_tx);
+    if (request == nullptr) {
+        LOG(ERROR) << "Failed building message cACTION_CONTROL_CHANNEL_SCAN_TRIGGER_SCAN_REQUEST";
+        return beerocks::eChannelScanErrCode::CHANNEL_SCAN_INTERNAL_FAILURE;
+    }
+
+    int32_t dwell_time_msec =
+        database.get_channel_scan_dwell_time_msec(m_radio_mac, m_is_single_scan);
+    if (dwell_time_msec < 0) {
```

Comment:
This should be `<= 0` IMO.

Reply:
I understand why, but keep in mind that the only way to reach this code is if the dwell time `!= 0`
Changed to `<= 0` anyway